source/DG.HostApp/Services/ClusterConfigActualizer.cs

```diff
@@ -0,0 +1,120 @@
+                            var clusterInstanceDescription = applicationInstances.First(app =>
+                                instance.Key == ApplicationExtensions.ConstructUniqueId(app.Type, app.Name));
+
+                            // case when instance was deleted from cluster
+                            if (clusterInstanceDescription == null)
+                            {
+                                this.applicationOrchestrator.Stop(instance.Key);
+                                this.applicationOrchestrator.UnRegister(instance.Key);
+                                continue;
+                            }
+
+                            // case when instance was moved to another host
+                            if (clusterInstanceDescription.PlacementPolicies.First(pp => pp == currentHost.Name) == null)
+                            {
+                                this.applicationOrchestrator.Stop(instance.Key);
+                                this.applicationOrchestrator.UnRegister(instance.Key);
+                                continue;
+                            }
+
+                            // case when amount of instance was changed
+                            int nodeInstanceAmountDiference = this.InstanceDifferenceCalculator(clusterInstanceDescription, currentHost, instance.Value.Count);
+
+                            if (nodeInstanceAmountDiference > 0)
+                            {
+                                this.applicationOrchestrator.BuildInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, clusterInstanceDescription.Settings?.ToString(), nodeInstanceAmountDiference);
+                            }
+                            else if (nodeInstanceAmountDiference < 0)
+                            {
+                                this.applicationOrchestrator.RemoveInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, Math.Abs(nodeInstanceAmountDiference));
+                            }
+                        }
+
+                        // starting the required applications
+                        foreach (var application in applicationInstances)
+                        {
+                            var isInstanceExist = instances.ContainsKey(ApplicationExtensions.ConstructUniqueId(application.Type, application.Name));
+                            var isHostedonThisNode = application.PlacementPolicies.First(x => x == currentHost.Name).Any();
+
+                            if (isHostedonThisNode && !isInstanceExist)
+                            {
+                                int nodeInstanceAmount = this.InstanceDifferenceCalculator(application, currentHost, 0);
+
+                                this.applicationOrchestrator.BuildInstance(application.Type, application.Name, application.Settings?.ToString(), nodeInstanceAmount);
+                            }
+                        }
+
+                        Thread.Sleep(5000);
```

Comment:
await Task.Delay, because Thread.Sleep is more lower level that TPL.

Reply:
Think about DelayService - if we use this code in unit tests - we have to wait at least 5 seconds to wait completion of this test